src/core/mapper.rs

```diff
@@ -0,0 +1,114 @@
+use crate::core::{
+    backend::core::BackendOperation,
+    configs::cluster::ClusterConfig,
+    configs::node::{DiskPath as ConfigDiskPath, NodeConfig},
+    data::*,
+};
+
+#[derive(Debug, Clone)]
+pub struct VDiskMapper {
+    local_node_name: String,
+    disks: Vec<DiskPath>,
+    vdisks: Vec<VDisk>,
+    nodes: Vec<Node>,
+}
+
+impl VDiskMapper {
+    pub fn new(vdisks: Vec<VDisk>, config: &NodeConfig, cluster: &ClusterConfig) -> VDiskMapper {
+        let (nodes, vdisks) = Self::prepare_nodes(vdisks, &cluster);
+
+        VDiskMapper {
+            vdisks,
+            local_node_name: config.name.as_ref().unwrap().to_string(),
+            disks: config
+                .disks()
+                .iter()
+                .map(|d| DiskPath::new(&d.name.clone(), &d.path.clone()))
+                .collect(),
+            nodes,
+        }
+    }
+    pub fn new2(
```

Comment:
what is `new2`??

Reply:
rename) it used when you run cluster from script and must run node with target name (not from node.yaml config)